Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -538,16 +538,31 @@ extension BrowserViewController: WKNavigationDelegate {
         }
 
         if !(url.scheme?.contains("firefox") ?? true) {
-            showSnackbar(forExternalUrl: url, tab: tab) { isOk in
-                guard isOk else { return }
-                UIApplication.shared.open(url, options: [:]) { openedURL in
-                    // Do not show error message for JS navigated links or redirect as it's not the result of a user action.
-                    if !openedURL, navigationAction.navigationType == .linkActivated {
-                        let alert = UIAlertController(title: .UnableToOpenURLErrorTitle, message: .UnableToOpenURLError, preferredStyle: .alert)
-                        alert.addAction(UIAlertAction(title: .OKString, style: .default, handler: nil))
-                        self.present(alert, animated: true, completion: nil)
+            func cancelHandlerAndLoadInWebview() {
```

Comment:
Just a quick comment, I'm not a big fan of `func` declarations in other `func`. They can be useful, but this delegate call is already quite big so if we could simplify a bit would be a great win!